### PR TITLE
s3limit

### PR DESCRIPTION
**Pull request by bahalool**
This change is [Reviewable](https://reviewable.io/reviews/kube-hpc/s3-adapter.hkube/1)

### CODE REVIEW

File: lib/s3-adapter.js [hunk 1]
```diff
@@ -12,16 +13,22 @@ class S3Adapter {
             S3Client.init(connection);
             this._wasInit = true;
             this._log = log;
+            S3Client.createBucket({ Bucket: 'hkube' });
```
Comment (NassiHarel) [inline lib/s3-adapter.js]: add await

File: lib/s3-adapter.js [hunk 2]
```diff
@@ -12,16 +13,22 @@ class S3Adapter {
...
+            S3Client.createBucket({ Bucket: 'hkube-results' });
         }
     }
 
     async putResults(options) {
-        return this.put({ jobId: options.jobId, taskId: 'result.json', data: options.data });
+        return this._put({ Bucket: 'hkube-results', Key: `${moment().format(DATE_FORMAT)}/${options.jobId}/result.json`, Body: options.data });
```
Comment (NassiHarel) [inline lib/s3-adapter.js]: Buckets should be consts and Promise.all to create buckets

**Comment by bahalool** [comment]
Review status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @NassiHarel)

---

*[lib/s3-adapter.js, line 16 at r1](https://reviewable.io/reviews/kube-hpc/s3-adapter.hkube/1#-LEiipsfdlsEEuhVYk1T-r1-16:-LEiuElZ4Cmtpga2zhY9:b-896fix) ([raw file](https://github.com/kube-hpc/s3-adapter.hkube/blob/a809c4bdc6ab46ede9b11685c1a7f3251b3c2332/lib/s3-adapter.js#L16)):*
<details><summary><i>Previously, NassiHarel (Nassi Harel) wrote…</i></summary><blockquote>

add await
</blockquote></details>

Done.

---

*[lib/s3-adapter.js, line 22 at r1](https://reviewable.io/reviews/kube-hpc/s3-adapter.hkube/1#-LEiipsfdlsEEuhVYk1T-r1-22:-LEiuFosBEy7N_Y3fb4X:b-896fix) ([raw file](https://github.com/kube-hpc/s3-adapter.hkube/blob/a809c4bdc6ab46ede9b11685c1a7f3251b3c2332/lib/s3-adapter.js#L22)):*
<details><summary><i>Previously, NassiHarel (Nassi Harel) wrote…</i></summary><blockquote>

Buckets should be consts and Promise.all to create buckets
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kube-hpc/s3-adapter.hkube/1)*

**Comment by bahalool** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed, 2 unresolved discussions (waiting on @NassiHarel)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kube-hpc/s3-adapter.hkube/1)*

File: lib/s3-adapter.js [hunk 3]
```diff
@@ -12,16 +14,23 @@ class S3Adapter {
             S3Client.init(connection);
             this._wasInit = true;
             this._log = log;
+            Object.values(BUCKETS_NAMES).forEach((bucketName) => {
```
Comment (NassiHarel) [inline lib/s3-adapter.js]: Add options params (bootstrap: true)
Change to Promise.all

Comment (NassiHarel) [inline lib/s3-adapter.js]: Add options params (bootstrap: true)
Change to Promise.all

**Pull request merged**